src/components/20-molecules/datepicker/ui.test.js

```diff
@@ -155,3 +162,72 @@ test('should render datepicker as reactified component', async t => {
   );
   await t.expect(datepickerReact.exists).ok();
 });
+
+// React controlled-component test
+fixture('Datepicker React controlled').page(
+  `${host}/iframe.html?id=molecules-datepicker-react-demos--controlled-component-react-ified-datepicker-with-inputfield`
+);
+test('datepicker should behave correctly when controlled', async t => {
+  const datepickerReact = await Selector(() =>
+    document.querySelector(
+      `axa-datepicker[data-test-id="datepicker-controlled-react"]`
+    )
+  );
+  await t.expect(datepickerReact.exists).ok();
+
+  const getInputValue = ClientFunction(
+    () =>
+      document.querySelector(
+        `axa-datepicker[data-test-id="datepicker-controlled-react"]`
+      ).value
+  );
+
+  await t.expect(await getInputValue()).eql('4.6.2019');
+
+  await t.typeText(
+    `axa-datepicker[data-test-id="datepicker-controlled-react"] .js-datepicker__input`,
+    '28.2.2019',
+    { replace: true }
+  );
+
+  await t.expect(await getInputValue()).eql('28.2.2019');
+  const checkbox = Selector(
+    'axa-checkbox[name="datepicker-controlled-react-checkbox"] .a-checkbox__wrapper'
+  );
+
+  await t.click(checkbox);
+
+  await t
+    .expect(
+      (await Selector('[data-test-id="datepicker-react-controlled-value"]'))
+        .innerText
+    )
+    .eql('28.2.2019 (frozen)');
+
+  await t.typeText(
+    `axa-datepicker[data-test-id="datepicker-controlled-react"] .js-datepicker__input`,
+    '5.6.2019',
+    { replace: true }
+  );
+
+  await t.expect(await getInputValue()).eql('28.2.2019');
+});
+
+fixture('Datepicker Form').page(
+  `${host}/iframe.html?id=molecules-datepicker-demos--feature-datepicker-in-a-form`
+);
+test('should submit datepicker correctly in form', async t => {
+  const datepickerForm = await Selector(() =>
+    document.querySelector(`axa-datepicker[data-test-id="datepicker-forms"]`)
+  );
+  await t.expect(datepickerForm.exists).ok();
+  await t.typeText(
+    `axa-datepicker[data-test-id="datepicker-forms"] .js-datepicker__input`,
+    '29.2.2020'
+  );
+  await t.click('#datepicker-forms-submit');
+  await t
+    .wait(50)
+    .expect((await Selector('#datepicker-forms-content')).innerText)
```

Comment:
Is this wait statement really needed? Testcafe anyway has a build in timeout to wait for elements to appear.

Reply:
Good question. After the click logic runs which takes time, the innerText will not appear instantaneously. That motivates the wait - note that it's not about the element which needs to exist, but its text which has to have a certain expected value. With that background info, is it OK for you?

Reply:
I will insert a comment to that effect into the code

Reply:
done